packages/amplify-cli-core/src/amplify-node-pkg-detector/amplify-nodejs-detector.ts

```diff
@@ -0,0 +1,106 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { AmplifyError } from '../errors/amplify-error';
+import { AmplifyFault } from '../errors/amplify-fault';
+import { getPackageManager, PackageManager } from '../utils/packageManager';
+import { Lockfile, LockfileParser } from './lock-file-interface';
+import { PackageJson } from './lock-file-types';
+import { LockFileParserFactory } from './parser-factory';
+
+/**
+ * return type of detectAffectedDirectDependencies
+ */
+export type DetectedDependency = {
+  packageName?: string;
+  dependentPackage?:{
+    name: string,
+    version: string,
+  }
+}
+
+/**
+ * props required by amplify detector
+ */
+export type AmplifyNodePkgDetectorProps = {
+    projectRoot: string,
+  }
+
+/**
+ * Amplify Node lock file dependency detector
+ * detects packages depending on pkg passed to Engine
+ * by parsing lock files
+ */
+export class AmplifyNodePkgDetector {
+     private readonly packageManager: PackageManager;
+     private readonly pkgJsonObj: PackageJson;
+     private readonly lockFileContents: string;
+     private readonly lockFileParser: LockfileParser;
+
+     constructor(amplifyDetectorProps: AmplifyNodePkgDetectorProps) {
+       const packageManager = getPackageManager(amplifyDetectorProps.projectRoot);
+       if (packageManager === null) {
+         throw new AmplifyError('MissingOverridesInstallationRequirementsError', {
+           message: 'No package manager found.',
+           resolution: 'Install npm or yarn to compile overrides for this project.',
+         });
+       }
+       this.packageManager = packageManager;
+       this.pkgJsonObj = this.parsePkgJson(amplifyDetectorProps.projectRoot);
+       this.lockFileContents = this.getLockFileContent(amplifyDetectorProps.projectRoot);
+       this.lockFileParser = LockFileParserFactory.getLockFileParser(this.packageManager.packageManager);
+     }
+
+     /**
+      * parses lock file
+      */
+     public parseLockFile():Lockfile {
+       return this.lockFileParser.parseLockFile(this.lockFileContents);
+     }
+
+     /**
+    * parses package.json project files
+    */
+     private parsePkgJson = (projectRoot: string): PackageJson => {
+       const pkgJsonFullPath = path.resolve(projectRoot, 'package.json');
+       return <PackageJson>JSON.parse(fs.readFileSync(pkgJsonFullPath, 'utf-8'));
+     }
+
+     /**
+      * get lock file content as string
+      */
+     private getLockFileContent = (projectRoot: string) : string => {
+       const lockFileFullPath = path.resolve(projectRoot, this.packageManager.lockFile);
+       if (!fs.existsSync(lockFileFullPath)) {
+         throw new AmplifyFault('LockFileNotFoundFault', {
+           message: 'Lockfile not found at location: ${lockFileFullPath}',
+         });
```

Comment:
Should this be error?

Reply:
We do a `npm install / yarn install `before calling this , so lock file should be there if there is `package.json` ( for this we have checks for package managers.)

So I think this should be fault as lock file should be present according to workflow.

Thoughts?